packages/cadl-ranch-specs/http/models/property-optional/mockapi.ts

```diff
@@ -0,0 +1,61 @@
+import { passOnSuccess, ScenarioMockApi, mockapi, json, MockApi } from "@azure-tools/cadl-ranch-api";
+
+export const Scenarios: Record<string, ScenarioMockApi> = {};
+
+interface MockApiGetPut {
+  get: MockApi;
+  put: MockApi;
+}
+
+/**
+ * Return the get and put operations
+ * @param route The route within /models/properties/optional/all/ for your function.
+ * @param body The body you are expecting and will return.
+ */
+function createMockApis(route: string, body: any): MockApiGetPut {
+  const url = `/models/properties/optional/${route}`;
+  console.log(url);
+  return {
+    get: mockapi.get(url, (req) => {
+      return {
+        status: 200,
+        body: json(body),
+      };
+    }),
+    put: mockapi.put(url, (req) => {
+      req.expect.bodyEquals(body);
+      return {
+        status: 204,
+      };
+    }),
+  };
+}
+
+const allOptionalRequiredAndOptional = createMockApis("all/requiredAndOptional", {
+  optionalString: "hello",
+  optionalInt: 42,
+  optionalStringList: ["hello", "world"],
+  optionalIntList: [1, 2],
+});
+Scenarios.Optional_AllOptional_get = passOnSuccess(allOptionalRequiredAndOptional.get);
+Scenarios.Optional_AllOptional_put = passOnSuccess(allOptionalRequiredAndOptional.put);
+
+const allOptionalRequiredOnly = createMockApis("all/requiredOnly", {});
+Scenarios.Optional_AllOptional_get_required_only = passOnSuccess(allOptionalRequiredOnly.get);
```

Comment:
@timotheeguerin based off of the path parameter, I have two scenarios per function. However, based off of the function name, there is just one required scenario. I went ahead and created my own scenarios, is this the correct way to do this?